### PR TITLE
General fixes

### PR DESCRIPTION
**Pull request by grant-d**
- Change `== null` to `is null`
- FxCop
- Nuget

### CODE REVIEW

**Comment by grant-d** [comment]
@jcdickinson most of the changes are straight-forward. Can you take a specific look at the 'FxCop' commit and let me know if those changes are OK.

File: src/SourceCode.Clay.OpenApi/Serialization/OasSerializer.Serialize.cs [hunk 1]
```diff
@@ -693,7 +693,7 @@ protected virtual JToken SerializeSchema(OasSchema value)
         /// <returns>The serialized value.</returns>
         public virtual JToken Serialize<T>(T value)
         {
-            if (ReferenceEquals(value, null)) return null;
+            if (Equals(value, default(T))) return null;
```
Comment (jcdickinson) [inline src/SourceCode.Clay.OpenApi/Serialization/OasSerializer.Serialize.cs]: False negative.

File: src/SourceCode.Clay.OpenApi/Serialization/OasSerializer.SerializeUnknown.cs [hunk 2]
```diff
@@ -110,7 +110,7 @@ public JToken Serialize<T>(OasSerializer serializer, T value)
         /// <returns>The serialized <see cref="JToken"/>.</returns>
         protected virtual JToken SerializeUnknown<T>(T value)
         {
-            if (ReferenceEquals(value, null)) return null;
+            if (Equals(value, default(T))) return null;
```
Comment (jcdickinson) [inline src/SourceCode.Clay.OpenApi/Serialization/OasSerializer.SerializeUnknown.cs]: False negative.

**Comment by grant-d** [comment]
Please contrib your preferred change on this PR directly. T can be a class or a struct.

**Comment by jcdickinson** [comment]
@grant-d I'm not getting a warning for reverting that change.

> ReferenceEquals(ValueType, null) is elided by the jitter, whereas Equals is not.

**Pull request merged**